node/events-to-ga/src/utils.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Throws an error if any of the keys are missing from the object
+ * @param {*} obj
+ * @param {string[]} keys
+ * @throws {Error}
+ */
+export function throwIfMissing(obj, keys) {
+  const missing = [];
+  for (let key of keys) {
+    if (!(key in obj) || !obj[key]) {
+      missing.push(key);
+    }
+  }
+  if (missing.length > 0) {
+    throw new Error(`Missing required fields: ${missing.join(", ")}`);
+  }
+}
+
+/**
+ * returns whether the request headers and appropriate message are ok
+ * @param {*} req
+ * @returns {{message:string,containsReqHeaders:boolean}}
+ */
+export function verifyHeaders(req) {
+  let containsReqHeaders = true;
+  let message = "";
+  if (req.headers["x-appwrite-user-id"] == "") {
+    message = `x-appwrite-user-id value in req.headers is not there`;
+    containsReqHeaders = false;
+  } else if (req.headers["x-appwrite-trigger"] != "event") {
+    message = `Not triggered by event but by ${req.headers["x-appwrite-trigger"]}`;
+    containsReqHeaders = false;
+  } else if (req.headers["x-appwrite-event"] == "") {
+    message = `x-appwrite-event value in req.headers is not there`;
+    containsReqHeaders = false;
+  }
+  return { message, containsReqHeaders };
+}
+
+/**
+ * returns formatted string by converting appwrite event name to Google Analytics accepted string
+ * @param {string} str
+ * @returns {{event_name:string,wildCardObject:{[key:string]:string}}}
+ */
+export function formatIntoGoogleAnalyticsEvent(str) {
+  const oddElemArray = [];
+  const wildCardArray = [];
+  const splitArray = str.split(".");
+
+  for (let i = 0; i < splitArray.length; i++) {
+    if (i % 2 == 0) oddElemArray.push(splitArray[i]);
+    else wildCardArray.push([`wildCard_${(i + 1) >> 1}`, splitArray[i]]);
+  }
+  const event_name = oddElemArray.join("_");
+  return {
+    event_name: event_name.charAt(0).toUpperCase() + event_name.slice(1),
+    wildCardObject: Object.fromEntries(wildCardArray),

```

Comment:
it would really help to have more descriptive keys rather than having the keys be wildCard_0, etc. It would be great if the key was actually what the value is. ie. databaseId, collectionId, etc.